qiskit/circuit/__init__.py

```diff
@@ -54,81 +78,825 @@
+
+    You turn an abstract circuit into a *physical circuit* by using :ref:`Qiskit's transpilation
+    package <qiskit-transpiler>`.
+
+circuit
+    A computational routine the defines a single execution to be taken on a QPU.  This can either be
+    an *abstract circuit* or a *physical circuit*.
+
+gate
+    A *unitary operation* on one or more qubits.
+
+hardware qubit
+    The representation of a single qubit on a particular *QPU*.  A hardware qubit has some physical
+    quantum-mechanical system backing it; unlike a *virtual qubit*, it has particular coupling
+    constraints and only certain gates can be applied to certain physical qubits.
+
+    Qiskit does not distinguish *logical qubits* from any individual *physical qubits* when talking
+    about hardware qubits.  A QPU may implement its hardware qubits as logical qubits, where each
+    hardware qubit comprises many physical qubits that are controlled and error-corrected opaquely
+    to Qiskit by the control electronics.  More likely, for near-term applications, a QPU will be
+    directly exposing its physical qubits as the hardware qubits for Qiskit to reason about.
+
+    Both physical and logical qubits will have coupling constraints between them, only permit
+    certain quantum operations on them, and have scheduling concerns between them.  Qiskit abstracts
+    these concerns together in the concept of hardware qubits. In the early days of quantum error
+    correction, particular backends may let you access their qubit resources either as high-level
+    logical qubits or as low-level physical qubits through Qiskit.
+
+instruction set architecture (ISA)
+    The abstract model of which operations are available on which sets of *hardware qubits* on one
+    particular *QPU*.  For example, one QPU may allow :math:`\sqrt X` and :math:`R_Z` operations on
+    all single hardware qubits, and :math:`CX` operations on certain pairs of hardware qubits.
+
+logical qubit
+    A collection of several *physical qubits* that are controlled together by a QPU (from the user's
+    perspective) to apply real-time quantum error correction.  A logical qubit is a type of
+    *hardware qubit* for Qiskit.
+
+measurement
+    The act of extracting one classical bit of a data from a single qubit state.  This is an
+    irreversible operation, and usually destroys entanglement and phase coherence between the target
+    qubit and the rest of the system.
+
+near-time classical computation
+    Any classical computation that can happen within the execution of a single shot of a *circuit*,
+    where the results of the classical computation can affect later execution of the circuit.  The
+    amount of near-time classical computation available with particular *QPU*\ s will vary
+    significantly dependent on many factors, such as the controlling electronics and the qubit
+    technology in use. You should consult your hardware vendor's documentation for more information
+    on this.
```

Comment:
I suggest to replace "near-term" with "at-runtime". I was confused when you used near-time compute in the introduction at first - here it becomes clear. :-)

Reply:
In theory, that's kind of the point of this definition section. That said, if you think "runtime" is going to be clear and not easily mixed up with "runtime of the Python program" or stuff like that, I can look at changing it everywhere?

Reply:
I'm not sure... The terminology offline/online/at-runtime also occurs in computer architecture but I agree that at-runtime might conflict with the python runtime and also with qiskit runtime. What do you think about "mid-circuit" or "in-circuit". What does 'near' refer to in 'near-time'? :-)

Reply:
Let's bring it up in the open qiskit-dev meeting and ask what terminology we want for this.

Reply:
From the dev meeting: the terminology from elsewhere in IBM Quantum is "real time", we think - I hadn't even considered that, but now I hear it I think it's the best choice.  Changed over in dd78e25a6c8.